File: LeetCode/RomanToInteger.java
class Solution {
    public static int romanToInt(String s) {
        int sum = 0;
        for (int i = 0; i < s.length(); i++) {
            if (String.valueOf(s.charAt(i)).equals("I")){
                if (i + 1 < s.length()) {
                    if (String.valueOf(s.charAt(i + 1)).equals("V")) {
                        sum += 4;
                        i++;
                    }
                    else if (String.valueOf(s.charAt(i + 1)).equals("X")) {
                        sum += 9;
                        i++;
                    }
                    else{
                        sum += 1;                    
                    }
                }
                else{
                        sum += 1;                    
                }
                
            }
            else if (String.valueOf(s.charAt(i)).equals("V")){
                sum += 5;                
            }
            else if (String.valueOf(s.charAt(i)).equals("X")){
                if (i + 1 < s.length()) {
                    if (String.valueOf(s.charAt(i + 1)).equals("L")) {
                        sum += 40;
                        i++;
                    }
                    else if (String.valueOf(s.charAt(i + 1)).equals("C")) {
                        sum += 90;
                        i++;
                    }
                    else{
                        sum += 10;                    
                    }
                }
                else{
                    sum += 10;                    
                }
            }
            else if (String.valueOf(s.charAt(i)).equals("L")){
                sum += 50;
            }
            else if (String.valueOf(s.charAt(i)).equals("C")){
                if (i + 1 < s.length()) {
                    if (String.valueOf(s.charAt(i + 1)).equals("D")) {
                        sum += 400;
                        i++;
                    }
                    else if (String.valueOf(s.charAt(i + 1)).equals("M")) {
                        sum += 900;
                        i++;
                    }
                    else{
                        sum += 100;                    
                    }
                }
                else{
                    sum += 100;                    
                }
            }
            else if (String.valueOf(s.charAt(i)).equals("D")){
                sum += 500;              
            }
            else if (String.valueOf(s.charAt(i)).equals("M")){
                sum += 1000;
                
            }
        }
        return sum;
    }
}

File: LeetCode/RomanToInteger.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        sum = 0
        special = ["I", "X", "C"] 
        # normal = ["V", "L", "D", "M"]
        # order = ["I", "V", "X", "L", "C", "D", "M"]
        # values = [1, 5, 10, 50, 100, 500, 1000]
        roman_dict = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
        for i in range(len(s)):
            if s[i] in special:
                if i + 1 != len(s):
                    if (s[i] == "I" and s[i + 1] == "V") or (s[i + 1] == "X" and s[i] == "I"):
                        add = roman_dict.get(s[i])
                        sum = sum - add
                    elif (s[i] == "X" and s[i + 1] == "L") or (s[i + 1] == "C" and s[i] == "X"):
                        add = roman_dict.get(s[i])
                        sum = sum - add
                    elif (s[i] == "C" and s[i + 1] == "D") or (s[i + 1] == "M" and s[i] == "C"):
                        add = roman_dict.get(s[i])
                        sum = sum - add
                    else:
                        add = roman_dict.get(s[i])
                        sum = sum + add 
                else:
                    add = roman_dict.get(s[i])
                    sum = sum + add 
            else:
                add = roman_dict.get(s[i])
                sum = sum + add
        return sum
    
s = Solution()
print(s.romanToInt("MCMXCIV"))